libpod/options.go

```diff
@@ -293,6 +293,17 @@ func WithHooksDir(hooksDirs ...string) RuntimeOption {
 	}
 }
 
+// WithCDI sets the devices to check for for CDI configuration.
+func WithCDI(devices []string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return define.ErrCtrFinalized
+		}
+		ctr.cdiDevices = devices
```

Comment:
This won't survive the initial start of the container, or even the first start if you use `podman create`

Reply:
Hey @mheon, isn't this the same pattern for every other config function in here? How can it survive?

Reply:
Thank you @mheon, moved cdiDevices to `ctr.config` for permanence as per your recommendation.